server/routes/cookies/CookiesPolicyController.ts

```diff
@@ -0,0 +1,53 @@
+import { NextFunction, Request, Response } from 'express'
+
+export default class CookiesPolicyController {
+  async initialiseCookieSession(req: Request, res: Response, next: NextFunction): Promise<void> {
+    if (!req.session.cookiesPolicy) {
+      req.session.cookiesPolicy = { policy: undefined, showConfirmation: false, lastPage: req.originalUrl }
+    }
+
+    if (res.locals.externalUser) {
+      req.session.cookiesPolicy.policy = req.cookies && req.cookies.cookies_policy
+      req.session.cookiesPolicy.showConfirmation = req.query.showCookieConfirmation === 'true'
+      if (!req.originalUrl.includes('cookies-policy')) {
+        req.session.cookiesPolicy.lastPage = req.originalUrl.replace('?showCookieConfirmation=true', '')
+      }
+    } else {
+      req.session.cookiesPolicy = { policy: 'n/a', showConfirmation: false, lastPage: req.originalUrl }
+    }
+
+    res.locals.cookiesPolicy = req.session.cookiesPolicy
+    res.locals.url = req.originalUrl
+    return next()
+  }
+
+  async getCookiesPolicyView(req: Request, res: Response): Promise<void> {
+    res.render('pages/cookies/cookies-policy', {
+      cookies: req.session.cookiesPolicy.policy,
+    })
+  }
+
+  async submitCookiesPolicyPreferences(req: Request, res: Response): Promise<void> {
+    let redirectUrl = req.session.cookiesPolicy.lastPage
+    if (req.originalUrl === '/cookies-policy') {
+      redirectUrl = '/cookies-policy'
+    }
+    if (req.body.cookies) {
+      if (req.body.cookies === 'reject') {
+        Object.entries(req.cookies).forEach(cookie => {
+          if (typeof cookie[0] === 'string' && /^_g/.test(cookie[0])) {
```

Comment:
Took me a while to grok this - is it basically saying "if you opted to reject cookies, we iterate through all the request cookies, and for each one, it element 0 (it's name ??) is a string and begins with _g (one of the google analytics cookies?), then we delete it.
Is that right?
I wonder if a comment, or refactoring into a method like "deleteGoogleAnaluyticsCookies" might help?

Reply:
Good call - I pinched that code from somewhere